pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -664,10 +664,12 @@ func TestDryRunPreemption(t *testing.T) {
 				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
 			)
 			registeredPlugins = append(registeredPlugins, tt.registerPlugins...)
+			informerFactory := informers.NewSharedInformerFactory(clientsetfake.NewSimpleClientset(), 0)
 			fwk, err := st.NewFramework(
 				registeredPlugins,
 				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
 				frameworkruntime.WithSnapshotSharedLister(snapshot),
+				frameworkruntime.WithInformerFactory(informerFactory),
```

Comment:
why is this necessary now?

Reply:
Because when default spreading is enabled, we require the informers for Services and so on.

Reply:
ok, and we were not using selectorspread in the tests (which would have required this).

Reply:
/hold
I didn't stop to think about that. Let me check

Reply:
/hold cancel

There is a test for preemption when there is hard topology spreading, which now has constraints by default.

Reply:
Yes, the test that registers PodTopologySpread plugin now has a system default constraints, and it can pass only with a non-nil informerFactory::

https://github.com/kubernetes/kubernetes/blob/af1c8935d9f1f42767c6f5211a668e44a15946b4/pkg/scheduler/framework/plugins/podtopologyspread/plugin.go#L96-L99